### ISSUE TITLE
The 1.0.0-beta.2 versioned package does not work with typescript < 4.1

### ISSUE THREAD

**Issue by aakashbhardwaj619** [Azure/communication-ui-library#619]
**Describe the bug; what happened?**
The **1.0.0-beta.2** versioned package does not work in a [SharePoint Framework](https://docs.microsoft.com/en-us/sharepoint/dev/spfx/sharepoint-framework-overview) project that uses TypeScript **3.9.7**. The **1.0.0-beta.1** version works fine in the same project.

**What are the steps to reproduce the issue?**
1. Create a new SharePoint Framework project ([Reference](https://docs.microsoft.com/en-us/sharepoint/dev/spfx/web-parts/get-started/build-a-hello-world-web-part)).
2. Add the **1.0.0-beta.2** version of azure/communication-react library.
3. Build the project and the errors mentioned in the screenshot below are received.

**What behavior did you expect?**
The project works fine with **1.0.0-beta.1** version so ideally, it should also work with **1.0.0-beta.2** unless there are any breaking changes.

**If applicable, provide screenshots:**
<img width="717" alt="ACSSPFxError" src="https://user-images.githubusercontent.com/33526902/126741877-7f098cb1-ad7f-4351-aed2-0a39599ff4e9.PNG">

**In what environment did you see the issue?**

- OS & Device: Windows
- Browser [e.g. Microsoft Edge
- Version 1.0.0-beta.2

**Comment by JamesBurnside** [comment]
Hi @aakashbhardwaj619, thanks for bringing this to our attention. We upgraded the api-extractor version in 1.0.0-beta.2 (this is what creates these .d.ts files). Looks like, because it now uses a newer version of typescript to generate these files, it needs our project to now declare a higher peer dependency version for typescript.

We'll look into what our solutions are (the upgraded api-extractor was to fix another issue so we will have to explore our options here). To unblock you are you able to upgrade to typescript 4.3.5?

**Comment by aakashbhardwaj619** [comment]
Thanks for acknowledging this @JamesBurnside. Unfortunately, the latest version of SharePoint Framework supports TypeScript only up to version 3.9. ([Reference](https://developer.microsoft.com/en-us/microsoft-365/blogs/announcing-sharepoint-framework-1-12-1-extending-more-of-microsoft-teams/))

**Comment by JamesBurnside** [comment]
We'll tackle this with two work items and have a fix ready for the next beta release:
1. @prprabhu-ms can you take a look at the chat error target type definition and see if you can make it typescript 3.9.7 compatible:
    ```
    /**
     * Helper type to build a string literal type containing methods of an object.
     */
    export type ChatObjectMethodNames<TName extends string, T> = {
      [K in keyof T]: `${TName}.${ChatMethodName<T, K>}`;
    }[keyof T];
    ```
    Using a string template in a type is only supported in typescript 4.1+
    * Perhaps we could construct the list of chatErrorTargets instead and export an enum of strings pertaining to the targets instead of the clever typing? (but still use the clever typing internally to guard against incorrectly named errors)
1. We need to build out some regression testing regarding older versions of typescript to ensure all type definitions are compatible with the minimum version of typescript we support

**Comment by JamesBurnside** [comment]
This is now fixed in 1.0.0-beta.3.
